File: __tests__/fixtures/kernel-parity/Torture.java
/**
 * Java torture fixture — exercises every Java extraction path the kernel
 * ports: package namespace, imports, javadoc, annotations, inheritance,
 * fields/constants, enums, anonymous classes, method references, static
 * member reads, fluent chains, Lombok synthesis, value refs + shadowing.
 */
package com.example.torture;

import java.util.List;
import java.util.Map;
import static java.util.Objects.requireNonNull;
import com.example.other.OtherClass;
import lombok.Data;

/** Javadoc for the service. */
@Service
@Component("torture")
public class TortureService extends BaseService implements Runnable, AutoCloseable {
  /** A shared constant table (value-ref target). */
  public static final Map<String, Integer> RETRY_LIMITS = Map.of("a", 1);
  private static final String API_BASE = "https://example.test";
  protected int count = 0;
  private final List<String> names;
  int packagePrivate, secondDeclarator;

  /** Ctor javadoc. */
  public TortureService(List<String> names) {
    this.names = requireNonNull(names);
    register(this::onEvent);
    queue(TortureService::compute);
    queue(OtherClass::handle);
    Runnable r = () -> helper(RETRY_LIMITS);
    executor.submit(new Runnable() {
      @Override
      public void run() {
        helper(RETRY_LIMITS);
      }
    });
  }

  @Override
  @Deprecated
  public void run() {
    Config cfg = ConfigLoader.getInstance().load();
    this.registry.lookup("x");
    helper(Direction.UP);
    String base = API_BASE;
    int max = Limits.MAX_VALUE;
    new StringBuilder(16).append(base);
  }

  private static Config helper(Object arg) {
    return new Config();
  }

  private void onEvent() {}
  private static void compute() {}

  public void shadowed() {
    String API_BASE = "local"; // shadows the class constant
    log(API_BASE);
  }

  enum Direction {
    UP,
    DOWN;

    Direction opposite() {
      return this == UP ? DOWN : UP;
    }
  }

  interface Listener {
    void onChange(TortureService svc);
  }
}

@Data
class LombokBean {
  private String name;
  private boolean isActive;
  private final int id;
  private static int counter;
  private String toString; // taken: no synthetic toString field collision

  public String getName() { return name; } // explicit getter — never overridden
}

@lombok.Getter
@lombok.extern.slf4j.Slf4j
@Builder
class LombokBuilderBean {
  private List<String> items;
}

interface Shape extends Comparable<Shape>, Cloneable {
  double area();
}

@interface Marker {
  String value() default "";
}

// Markdown path references: code -> documentation edges. Every shape the
// normalizer branches on, so the two arms have to agree about the rejections
// (URL, escape above the root) as well as the emissions.
class MarkdownPaths {
  static final String GUIDE = "../docs/guide.md#install";
  static final String BARE = "README.md";
  static final String ROOTED = "/docs/rooted.md";
  static final String ESCAPES = "../../../../outside.md";
  static final String REMOTE = "https://example.com/remote.md";
  static final String QUERIED = "./notes.md?raw=1#top";
  static final String TWO_IN_ONE = "see a.md and also sub/b.mdx";

  void load() {
    loadDoc("docs/deep/nested.markdown");
  }
}

File: __tests__/fixtures/kernel-parity/torture.go
// Go torture fixture — receivers, embedding, interfaces, composite literals.
package torture

import (
	"fmt"
	pkga "example.com/other/pkga"
)

const MAX_ITEMS = 128

var DefaultRegistry = NewRegistry()

var handlerTable = map[string]func(int){
	"recv": TargetCb,
}

type Widget struct {
	*Base
	Queryable
	name string
}

type Stack[T any] struct {
	items []T
}

type Core interface {
	Reader
	Marshal(v any) ([]byte, error)
	Unmarshal(data []byte) error
}

type Dur int

func NewRegistry() *Registry {
	w := Widget{name: "w"}
	q := pkga.Widget{}
	fmt.Println(w, q, MAX_ITEMS)
	cfg := loadConfig()
	cfg.conn.Exec("x")
	return New().Init()
}

func (s *Stack[T]) Push(item T) {
	s.items = append(s.items, item)
}

func (w Widget) Render() string {
	return w.name
}

func TargetCb(n int) {}

func shadowed() {
	MAX_ITEMS := 5
	fmt.Println(MAX_ITEMS)
}

func reads() int {
	return MAX_ITEMS
}

// Markdown path references: code -> documentation edges. Every shape the
// normalizer branches on, so the two arms have to agree about the rejections
// (URL, escape above the root) as well as the emissions.
var mdGuide = "../docs/guide.md#install"
var mdBare = "README.md"
var mdRooted = "/docs/rooted.md"
var mdEscapes = "../../../../outside.md"
var mdRemote = "https://example.com/remote.md"
var mdQueried = "./notes.md?raw=1#top"
var mdTwoInOne = "see a.md and also sub/b.mdx"

func mdLoad() {
	loadDoc("docs/deep/nested.markdown")
}
